packages/node_modules/@webex/internal-plugin-devices/src/devices.js

```diff
@@ -0,0 +1,795 @@
+    return Promise.reject(new Error(
+      '`wdm` service is not currently available'
+    ));
+  },
+
+  /**
+   * Check if the device can currently reach the inactivity check url.
+   *
+   * @returns {Promise<void, Error>}
+   */
+  checkNetworkReachability() {
+    this.logger.info('devices: checking network reachability');
+
+    // Validate if the device has been checked and reset the logout timer.
+    if (this.isReachabilityChecked) {
+      return Promise.resolve(this.resetLogoutTimer());
+    }
+
+    // Validate if the device has a intranet checking url.
+    if (!this.intranetInactivityCheckUrl) {
+      this.isInNetwork = false;
+
+      return Promise.resolve(this.resetLogoutTimer());
+    }
+
+    this.isReachabilityChecked = true;
+
+    // Clear unnecessary headers for reachability request.
+    const headers = {
+      'cisco-no-http-redirect': null,
+      'spark-user-agent': null,
+      trackingid: null
+    };
+
+    // Send the network reachability request.
+    return this.webex.request({
+      headers,
+      method: 'GET',
+      uri: this.intranetInactivityCheckUrl
+    })
+      .then(() => {
+        this.isInNetwork = true;
+
+        return Promise.resolve(this.resetLogoutTimer());
+      })
+      .catch(() => {
+        this.logger.info('devices: did not reach ping endpoint');
+        this.logger.info('devices: triggering off-network timer');
+
+        this.isInNetwork = false;
+
+        return Promise.resolve(this.resetLogoutTimer());
+      });
+  },
+
+  /**
+   * Clears the registration ttl value if available.
+   *
+   * @param {Object} options - Values to be cleared.
+   * @returns {void}
+   */
+  clear(...args) {
+    this.logger.info('devices: clearing registered device');
+
+    // Prototype the extended class in order to preserve the parent member.
+    Reflect.apply(WebexPlugin.prototype.clear, this, args);
+  },
+
+  /**
+   * Get the current websocket url with the appropriate priority host.
+   *
+   * @param {boolean} [wait] - Willing to wait on a valid url.
+   * @returns {Promise<string, Error>} - The priority-mapped web socket url.
+   */
+  getWebSocketUrl(wait = false) {
+    this.logger.info('devices: getting the current websocket url');
+
+    // Destructure the services plugin for ease of reference.
+    const {services} = this.webex.internal;
+    let url;
+
+    // Validate if the method should wait for registration.
+    if (wait) {
+      return this.waitForRegistration()
+        .then(() => {
+          // Attempt to collect the priority-mapped url.
+          try {
+            url = services.convertUrlToPriorityHostUrl(this.webSocketUrl);
+          }
+          catch (error) {
+            return Promise.reject(error);
+          }
+
+          return url;
+        })
+        .catch((error) => {
+          this.logger.warn(error.message);
+
+          return Promise.reject(new Error(
+            'devices: failed to get the current websocket url'
+          ));
+        });
```

Comment:
Based on how `convertUrlToPriorityHostUrl()` throws an error, I'm curious if there would be a better approach here, outside of try-catch.